openfl/component/director/experiment.py

```diff
@@ -189,7 +191,7 @@ def finish_active(self) -> None:
         self.__active_experiment_name = None
 
     @asynccontextmanager
-    async def get_next_experiment(self):
+    async def get_next_experiment(self) -> AsyncGenerator[Experiment, Any]:
```

Comment:
Let's annotate it according to:
`AsyncGenerator[yield_type, send_type, return_type] `

Reply:
Based on https://docs.python.org/3/library/typing.html#:~:text=class%20typing.AsyncGenerator,SendType%5D.%20For%20example%3A
I guess there should be `AsyncGenerator[Experiment, None]`